include/boot.h

```diff
@@ -56,6 +56,24 @@ typedef struct __packed lz_header {
 } lz_header_t;
 extern lz_header_t lz_header;
 
+#define MLE_UUID0	0x9082ac5a
+#define MLE_UUID1	0x74a7476f
+#define MLE_UUID2	0xa2555c0f
+#define MLE_UUID3	0x42b651cb
+typedef struct __packed mle_header {
+	u32 uuid[4];
+	u32 size;		/* 0x00000034 MLE header size */
+	u32 version;		/* 0x00020002 MLE version 2.2 */
+	u32 sl_stub_entry;	/* Linear entry point of MLE (virt. address) */
+	/* The following fields are used only for Intel TXT */
+	u32 first_page;		/* First valid page of MLE */
+	u32 start_offset;	/* Offset within binary of first byte of MLE */
+	u32 end_offset;		/* Offset within binary of last byte + 1 of MLE */
+	u32 vector;		/* Bit vector of MLE-supported capabilities */
+	u32 cmdline_start;	/* Starting linear address of command line */
```

Comment:
I could be wrong but is there a reason why AMD shouldn't also be passing the location of the kernel command line?

Reply:
It is passed in the zero page. I don't know why it is needed for Intel, then again I didn't look too deep into TXT. Perhaps it has something to do with protection against DMA?

Reply:
@rossphilipson, I am pretty certain we discussed this, do you recall why we put it in the SL kern_info structure?

Reply:
It is defined in the TXT spec and it is optional but IIRC we are not using it.

Reply:
okay, so based on @rossphilipson response,  @krystian-hebel you can ignore my comment.